### DISCUSSION TITLE
How to get code line numbers inside page margins?

### DISCUSSION THREAD

**Discussion by MohamedEmary** [jgm/pandoc#9647] [Q&A] [answered]
I'm using pandoc to generate PDFs from markdown files, when i write a code block with line numbers using `{.lang .numberLines}` the line numbers goes outside page margins like that:

![Code line numbers outside page margins](https://github.com/jgm/pandoc/assets/64380567/b03aaf27-c2c1-4d53-a855-42b04abd3339)

Is there anyway to fix that.

I tried searching for this issue but most of the answers [like this one](https://tex.stackexchange.com/a/135709) were for how to fix that behavior when writing pure latex not pandoc flavored markdown.

**Comment by jgm** [comment]
We're using a customized fancyvrb Verbatim environment, so see fancyvrb's documentation.

You could try the following diff in the default template:

```
- \DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
+ \DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\},xleftmargin=1.5em}
```

**Reply by jgm** [reply, marked as answer]
Sorry, forgot that that line is not in the template, because it is supplied by skylighting.
In  that case, you can try adding this to `header-includes`:
```latex
\RecustomVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}, xleftmargin=1.5em}
```

**Reply by MohamedEmary** [reply]
Thank you so much, this solved the problem!
